packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1107,6 +1090,86 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  private onBootCleanup() {
+    this.initApprovalsOfUnapprovedTransactions();
+    this.loadGasValuesOfUnapprovedTransactions();
+    this.submitApprovedTransactions();
+  }
+
+  /**
+   * Create approvals for all unapproved transactions on current chain.
+   */
+  private initApprovalsOfUnapprovedTransactions() {
+    const chainId = this.getChainId();
+    const unapprovedTransactions = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const transactionMeta of unapprovedTransactions) {
+      this.processApproval(transactionMeta, {
+        shouldShowRequest: false,
+      }).catch((error) => {
+        /* istanbul ignore next */
+        console.error('Error during persisted transaction approval', error);
+      });
+    }
+  }
+
+  /**
+   * Update the gas values of all unapproved transactions on current chain.
+   */
+  private async loadGasValuesOfUnapprovedTransactions() {
+    const chainId = this.getChainId();
+    const isEIP1559Compatible = await this.getEIP1559Compatibility();
+    const unapprovedTransactions = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const transactionMeta of unapprovedTransactions) {
```

Comment:
Should we await all these in parallel to save some time?

Reply:
Updated this to make this parallel in https://github.com/MetaMask/core/pull/1916/commits/2b3a6ae643e9427ab855eb2e3ebcbd50ab667987

Also added another transaction to the test case in order to check one estimated properly while another fails.